petastorm/reader.py

```diff
@@ -174,6 +174,24 @@ def make_reader(dataset_url,
                            'Inner exception: %s', str(e))
 
 
+def normalize_data_url(dataset_url, hdfs_driver):
+    if dataset_url is None or not isinstance(dataset_url, six.string_types):
+        raise ValueError('dataset_url must be a string')
+
+    try:
+        dataset_metadata.get_schema_from_dataset_url(dataset_url, hdfs_driver=hdfs_driver)
+        warnings.warn('Please use make_reader (instead of \'make_batch_dataset\' function to read this dataset. '
+                      'You may get unexpected results. '
+                      'Currently make_batch_reader supports reading only Parquet stores that contain '
+                      'standard Parquet data types and do not require petastorm decoding.')
+    except PetastormMetadataError:
+        pass
```

Comment:
Why swallow the exception ?
And the warning say irrelevant things..?

Reply:
I just refactored this part of the code from `make_batch_reader`. Shall we modify it?